### PR TITLE
Move canonical version numbers into source, not build system (#7980)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Review by derek-gerstmann — APPROVED** [review]

File: test/runtime/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,6 @@ function(_set_target_options NAME)
     target_compile_definitions(
         ${NAME}
         PRIVATE
-        HALIDE_VERSION=${Halide_VERSION}
         HALIDE_VERSION_MAJOR=${Halide_VERSION_MAJOR}
         HALIDE_VERSION_MINOR=${Halide_VERSION_MINOR}
         HALIDE_VERSION_PATCH=${Halide_VERSION_PATCH}
```
Comment (alexreinking) [inline test/runtime/CMakeLists.txt:13]: Should these not be removed, too?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -549,8 +543,10 @@ set_target_properties(Halide PROPERTIES
...
                       SOVERSION ${Halide_SOVERSION_OVERRIDE})
 
+# Note that we (deliberately) redeclare these versions here, even though the macros
+# with identical versions are expected to be defined in source; this allows us to
+# ensure that the versions defined between all build systems are identical.
 target_compile_definitions(Halide PUBLIC 
-                           HALIDE_VERSION=${Halide_VERSION} 
                            HALIDE_VERSION_MAJOR=${Halide_VERSION_MAJOR}
                            HALIDE_VERSION_MINOR=${Halide_VERSION_MINOR}
                            HALIDE_VERSION_PATCH=${Halide_VERSION_PATCH})
```
Comment (alexreinking) [inline src/CMakeLists.txt:552]: Why weren't these removed?

**Review by alexreinking — CHANGES_REQUESTED** [review]
Is this complete?

**Comment by alexreinking** [comment]
It would be nice if we had some helper script or the like to consistently update the version numbers and check that they're consistent.

File: src/CMakeLists.txt (hunk 2 above)
Comment (steven-johnson) [reply src/CMakeLists.txt:552]: As a way to verify that the CMake-defined verisions match the source-defined versions

**Review by alexreinking — APPROVED** [review]

File: src/CMakeLists.txt (hunk 2 above)
Comment (alexreinking) [reply src/CMakeLists.txt:552]: > As a way to verify that the CMake-defined verisions match the source-defined versions

Gotcha - I think I missed where that happens.

**Pull request merged**